pallets/order-book/src/tests.rs

```diff
@@ -380,6 +389,325 @@ fn fill_order_full_works() {
+				System::events()[5].event,
+				RuntimeEvent::OrderBook(Event::OrderFulfillment {
+					order_id,
+					placing_account: order.placing_account,
+					fulfilling_account: ACCOUNT_1,
+					partial_fulfillment: false,
+					fulfillment_amount: buy_amount,
+					currency_in: order.asset_in_id,
+					currency_out: order.asset_out_id,
+					sell_rate_limit: order.max_sell_rate,
+				})
+			);
+		});
+	}
+
+	#[test]
+	fn fill_order_partial_bad_origin() {
+		new_test_ext().execute_with(|| {
+			let buy_amount = 100 * CURRENCY_AUSD_DECIMALS;
+			let min_fulfillment_amount = 10 * CURRENCY_AUSD_DECIMALS;
+			let sell_ratio = FixedU128::checked_from_rational(3u32, 2u32).unwrap();
+
+			assert_ok!(OrderBook::place_order(
+				ACCOUNT_0,
+				DEV_AUSD_CURRENCY_ID,
+				DEV_USDT_CURRENCY_ID,
+				buy_amount,
+				sell_ratio,
+				min_fulfillment_amount,
+			));
+
+			let (order_id, _) = get_account_orders(ACCOUNT_0).unwrap()[0];
+
+			assert_noop!(
+				OrderBook::fill_order_partial(
+					RawOrigin::None.into(),
+					order_id,
+					min_fulfillment_amount,
+				),
+				BadOrigin
+			);
+		});
+	}
+
+	#[test]
+	fn fill_order_partial_invalid_order() {
+		new_test_ext().execute_with(|| {
+			assert_noop!(
+				OrderBook::fill_order_partial(
+					RuntimeOrigin::signed(ACCOUNT_1),
+					1234,
+					10 * CURRENCY_AUSD_DECIMALS,
+				),
+				Error::<Runtime>::OrderNotFound
+			);
+		});
+	}
+
+	#[test]
+	fn fill_order_partial_insufficient_order_size() {
+		new_test_ext().execute_with(|| {
+			let buy_amount = 100 * CURRENCY_AUSD_DECIMALS;
+			let min_fulfillment_amount = 10 * CURRENCY_AUSD_DECIMALS;
+			let sell_ratio = FixedU128::checked_from_rational(3u32, 2u32).unwrap();
+
+			assert_ok!(OrderBook::place_order(
+				ACCOUNT_0,
+				DEV_AUSD_CURRENCY_ID,
+				DEV_USDT_CURRENCY_ID,
+				buy_amount,
+				sell_ratio,
+				min_fulfillment_amount,
+			));
+
+			let (order_id, _) = get_account_orders(ACCOUNT_0).unwrap()[0];
+
+			assert_noop!(
+				OrderBook::fill_order_partial(
+					RuntimeOrigin::signed(ACCOUNT_1),
+					order_id,
+					min_fulfillment_amount - 1 * CURRENCY_AUSD_DECIMALS,
+				),
+				Error::<Runtime>::InsufficientOrderSize
+			);
+		});
+	}
+
+	#[test]
+	fn fill_order_partial_insufficient_asset_funds() {
+		new_test_ext().execute_with(|| {
+			let buy_amount = 100 * CURRENCY_AUSD_DECIMALS;
+			let min_fulfillment_amount = 1 * CURRENCY_AUSD_DECIMALS;
+			let sell_ratio = FixedU128::checked_from_rational(3u32, 2u32).unwrap();
+
+			assert_ok!(OrderBook::place_order(
+				ACCOUNT_0,
+				DEV_AUSD_CURRENCY_ID,
+				DEV_USDT_CURRENCY_ID,
+				buy_amount,
+				sell_ratio,
+				min_fulfillment_amount,
+			));
+
+			let (order_id, _) = get_account_orders(ACCOUNT_0).unwrap()[0];
+
+			let total_balance = OrmlTokens::balance(DEV_AUSD_CURRENCY_ID, &ACCOUNT_1);
+			assert_ok!(OrmlTokens::hold(
+				DEV_AUSD_CURRENCY_ID,
+				&ACCOUNT_1,
+				total_balance
+			));
+
+			// TODO(cdamian): No way of triggering `InsufficientAssetFunds` given that
```

Comment:
Reminder to double-check this.

Reply:
I also ran into this _issue_ during integration tests. AFAIR, I always encountered `BalanceTooLow`. Would be great if we fixed this for better debugging.

Reply:
Yes, I guess there is a bug in the `can_hold` logic that we have, gonna check it out.

Reply:
Fixed this cc @mustermeiszer

Reply:
Where? ^^ Can not find it.

Reply:
Just pushed, sorry.